src/gt4py/next/common.py

```diff
@@ -117,10 +129,102 @@ def __getitem__(self, index: int | slice) -> int | UnitRange:
             else:
                 raise IndexError("UnitRange index out of range")
 
+    def __and__(self, other: UnitRange) -> UnitRange:
+        start = max(self.start, other.start)
+        stop = min(self.stop, other.stop)
+        return UnitRange(start, stop)
+
 
 DomainT: TypeAlias = tuple[tuple[Dimension, UnitRange], ...]
 
 
+from dataclasses import dataclass
+from collections.abc import Sequence
+
+
```

Comment:
maybe we need something like 

```
NamedRange: TypeAlias = tuple[Dimension, UnitRange]
NamedPosition: TypeAlias = tuple[[Dimension, int], ...]
```

Reply:
So for now I just added `NamedRange` as this is what `Domain` will return from `__get_item__` and `__contains__`, as far as I understand we might worry about `NamedPosition` later?

Reply:
Makes sense!